mauth_client/utils.py

```diff
@@ -20,3 +21,14 @@ def hexdigest(val):
 
 def base64_encode(signature):
     return base64.b64encode(signature).decode("US-ASCII").replace("\n", "")
+
+
+def decode(byte_string: bytes) -> str:
+    """
+    Attempt to decode a byte string with utf and fallback to cchardet.
+    """
+    try:
+        return byte_string.decode("utf-8")
+    except UnicodeDecodeError:
+        encoding = cchardet.detect(byte_string)["encoding"]
+        return byte_string.decode(encoding)
```

Comment:
what do we think about this decoding? since this middleware is in low-level ASGI, we get the headers, query  string, and body as bytes. i don't think headers can have non-utf8 chars, but maybe query string and/or body? i went with  trying to detect the encoding if utf8 fails but that is not 100% accurate.  i think it is probably "good enough" though.

also makes me wonder if we thought about encodings in our protocol test suite.

Reply:
🤷 Assuming UTF8 seems reasonable to me but I agree this seems to be a gap in our thinking.